### PR TITLE
adds sign up feature test and fix bug for sign up

### PR DESCRIPTION
**Pull request by AlbertStjarne**
https://www.pivotaltracker.com/story/show/160580146

```
As as user,
In order connect to my friends,
I would like to sign up and get an account.
```

We added the features for the sign up flow from start of sign up to a successful sign up.
Step definitions were added for all of the sign up steps.

We learnt how to work outside in, in this case with the sign up process, how to write features and connected step definitions.

A bug was fixed on the sign up, find the bug report used below:

#### Expected behavior
- When signing up, the form is filled, then the new user should be saved.
#### Actual behavior
- Not allowing new user.
#### Steps to reproduce the problem
- Click sign up button
- Fill in the form
- Click sign up

### CODE REVIEW

**Comment by faraznaeem** [comment]
Please use markdown to format the PR description

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def conversation
   protected
 
   def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
```
Comment (faraznaeem) [inline app/controllers/application_controller.rb:23]: Please explain why you are adding this line

File: features/sign_up_to_the_app.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Scenario:
+Given I am on the landing page
+When I click on the link "Sign up"
+And I fill in the "Name" with "Albert"
+And I fill in the "Email" with "albert.stjarne@gmail.com"
+And I fill in the "Password" with "mypassword"
+And I fill in the "Password confirmation" with "mypassword"
+When I click on the link "Create"
+And I should see message "Welcome! You have signed up successfully."
```
Comment (faraznaeem) [inline features/sign_up_to_the_app.feature]: Please format the code, with proper indentation

File: features/step_definitions/sign_up_step_definitions_step.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+end
+
+
+
+# Then("the following user exists") do |table|
+#   table.hashes.each do |name|
+#     name.create!(name)
+#   end
+#  end
+
```
Comment (faraznaeem) [inline features/step_definitions/sign_up_step_definitions_step.rb]: always remove comments before submitting a PR
And format the code

**Review by faraznaeem — CHANGES_REQUESTED** [review]
You are missing unit tests for this feature, when you sign up for an account the user is saved in the DB and therefore you need to test if it is actually saved in the DB.

File: features/step_definitions/sign_up_step_definitions_step.rb (hunk 3 above)
Comment (tochman) [reply features/step_definitions/sign_up_step_definitions_step.rb]: At least when you submit it for final review.

Comment (AlbertStjarne) [reply features/step_definitions/sign_up_step_definitions_step.rb]: fixed

File: features/sign_up_to_the_app.feature (hunk 2 above)
Comment (AlbertStjarne) [reply features/sign_up_to_the_app.feature]: fixed

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (AlbertStjarne) [reply app/controllers/application_controller.rb:23]: The ParameterSanitizer deals with permitting specific parameters values for each Devise scope in the application.

The sanitizer knows about Devise default parameters (like password and password_confirmation for the `RegistrationsController`), and you can extend or change the permitted parameters list on your controllers.

File: features/sign_up_to_the_app.feature (hunk 2 above)
Comment (AlbertStjarne) [reply features/sign_up_to_the_app.feature]: fixed

File: spec/factories/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user do
+    name 'Albert'
+    email  'albert.stjarne@gmail.com'
+    password  'mypassword'
```
Comment (faraznaeem) [inline spec/factories/user.rb]: avoid using real emails

File: features/sign_up_to_the_app.feature [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  Given I am on the landing page
+  When I click on the link "Sign up"
+  And I fill in the "Name" with "John-Alexander"
+  And I fill in the "Email" with "john@alexander.com"
+  And I fill in the "Password" with "x"
+  And I fill in the "Password confirmation" with "y"
+  When I click on the link "Create"
+  And I should see message "Password confirmation doesn't match Password"
+  And I should see message "Password is too short (minimum is 8 characters)"
+  And I should see message "Name is too long (maximum is 10 characters)"
```
Comment (faraznaeem) [inline features/sign_up_to_the_app.feature]: Please go through the steps the steps `Given, When, Then and ... And`

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: spec/factories/user.rb (hunk 4 above)
Comment (AlbertStjarne) [reply spec/factories/user.rb]: fixed

File: features/sign_up_to_the_app.feature (hunk 5 above)
Comment (AlbertStjarne) [reply features/sign_up_to_the_app.feature]: changed the wording

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**